src/google/google-calendars/googlecalendarsyncadaptor.cpp

```diff
@@ -1342,6 +1342,11 @@ void GoogleCalendarSyncAdaptor::finishedRequestingRemoteEvents(int accountId, co
     foreach (const QString &updatedCalendarId, m_calendarsFinishedRequested.keys()) {
```

Comment:
I don't think so - the plugin instance lifetime is exactly one sync with exactly one account, so unless there is a logic bug in the plugin which makes us request a single calendar multiple times, for example, I don't think the map could contain bad data. 

What it may contain, however, is out-of-date data, if the notebook is deleted, or its pluginName field is modified, between when we requested the calendar data, and this point in the code - this guard should now protect from that case.